### PR TITLE
oauth2l: init at 1.2.2

### PR DESCRIPTION
**Pull request by pjjw**
###### Motivation for this change
The desire to use this tool in a development workflow.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by dbirks — APPROVED** [review]
Looks good to me! Double-checked the meta.homepage and license, and built and ran on NixOS:

Result of `nixpkgs-review pr 136199` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>oauth2l</li>
  </ul>
</details>

File: pkgs/tools/security/oauth2l/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, buildGoPackage, fetchFromGitHub, ... }:
+
+let version = "1.2.2"; in
+
+buildGoPackage {
```
Comment (dbirks) [inline pkgs/tools/security/oauth2l/default.nix:5]: Oh I'm sorry, I missed this. Can you change to `buildGoModule`? The nixpkgs documentation says `buildGoPackage` is the legacy way:
https://nixos.org/manual/nixpkgs/stable/#sec-language-go

**Review by dbirks — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/oauth2l/default.nix (hunk 1 above)
Comment (pjjw) [reply pkgs/tools/security/oauth2l/default.nix:5]: oops! sure, thanks for the catch- this absolutely was just a template pulled from the closest simplest go build.

thanks for paying attention!

Comment (pjjw) [reply pkgs/tools/security/oauth2l/default.nix]: having some issue with the test environment, will circle back when i've figured it out.

Comment (pjjw) [reply pkgs/tools/security/oauth2l/default.nix]: got it- had to stick in a bummer of a flag, but that seems the only fix to be made without modifying the source.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 136199` at 7f8a141a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/ef1887d8b14ca93abd9023494227798b">oauth2l</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Pull request closed without merging**